helper/jsonutil/json.go

```diff
@@ -17,6 +23,28 @@ func EncodeJSON(in interface{}) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// EncodeJSONAndCompress encodes the given input into JSON and compresses the
+// encoded value using Gzip format (BestCompression level). A canary byte is
+// placed at the beginning of the returned bytes for the logic in decompression
+// method to identify compressed input.
+func EncodeJSONAndCompress(in interface{}) ([]byte, error) {
```

Comment:
Can we allow users to pass in an optionally-nil compression config here? I don't think gzip best compression will necessarily be right for all uses as it's quite slow, but it seems like a decent default.